ADR/user-specified-tokens-build.md

```diff
@@ -0,0 +1,75 @@
+# Enable users to configure tokens/credentials for a Build/Test Pipeline
+
+Date: 2023-03-03
+
+## Status
+
+Provisional
+
+
+## Context
+
+As a user, I would like to configure sensitive information/tokens/credentials for the build/test pipeline to consume as part of a partner Task.
+
+## What is a Partner Task
+
+A Partner Task is a Tekton Task that is provided and maintained by a Red Hat Partner. However, this initiative also aims to enable Tekton Tasks 
+contributed to the StoneSoup Build/Test Pipeline by different organizations inside Red Hat.
+
+
+## Decision
+
+### How would the sensitive information be persisted and made available to the PipelineRun ?
+
+1. The UI would take in sensitive information as form input and upload it to SPI using the user's authentication token. See https://github.com/redhat-appstudio/service-provider-integration-operator/blob/main/docs/USER.md#uploading-access-token-to-spi-using-kubernetes-secret .
+
+2. Subsequently, the UI would create an `SPIAccessTokenBinding` CR with the appropriate label in the `.spec.secret.labels` to make the `Secrets` available to the build service or the integration service.
+
+
+### How does the build service and the integration service discover the `Secrets` ?
+
+While uploading a token, the user should be able to choose the name of the Pipeline `parameter` from a UI dropdown. The dropdown would be a pre-populated and hard-coded list of names associated with known partner integrations. 
+
+The user should also be allowed to provide a parameter name of choice. The same could be used in a customized PipelineRun definition in the user's source code's .tekton/ directory.
+
+| Pipeline Parameter    | Contents of the Secret | Name of the Secret        |
+| --------------------  | ---------------------- | ------------------------  |
+| codecov-github-token  | *input from the user*  | *autogenerated*           | 
+| codecov-token-secret  | *input from the user*  | *autogenerated*           |
+| redhat-ocm-token      | *input from the user*  | *autogenerated*           | 
+| *input from the user* | *input from the user*  | *autogenerated*           | 
+
+
+The build-service or the integration-service, reponsible for generating the PipelineRun needs to be able to map the name of the PipelineRun `parameter` with that of the `Secret` created by the UI, with contents provided by the user. To help the build-service do so, the `secrets` would be labelled with the name of the parameter for the build-service to look up, by the UI. 
+
+```
+apiVersion: appstudio.redhat.com/v1beta1
+kind: SPIAccessTokenBinding
+metadata:
+ name: doesnt-matter
+ namespace: user-tenant
+spec:
+ repoUrl: https://snyk.io/
```

Comment:
What should be the repoUrl? A fixed value for all secrets?

Reply:
Good question, repoURL doesn't really matter. @skabashnyuk ?

Reply:
So UI should use appropriate providers here based on the parameter selected?

Reply:
@Michkov, are you trying to ask if it would be supported for there to be a secret which does not have a repoURL (i.e. to be associated with a custom task were created which didn't specify that)?

Reply:
> Yeah - it just represents the provider. It has to match between the upload secret and the spiaccesstokenbinding in order to have the right one injected.

Right, but apart from it being useful to surface 'text', does it really matter ? Since the linking is anyway happening because of the name of the secret.